src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -157,15 +157,19 @@ public static string GetFileNameWithoutExtension(string path)
 
         // Returns a cryptographically strong random 8.3 string that can be 
         // used as either a folder name or a file name.
-        public static string GetRandomFileName()
+        public static unsafe string GetRandomFileName()
         {
             // 5 bytes == 40 bits == 40/5 == 8 chars in our encoding.
             // So 10 bytes provides 16 chars, of which we need 11
             // for the 8.3 name.
-            byte[] key = CreateCryptoRandomByteArray(10);
+            const int KeyLength = 10;
+            byte* key = stackalloc byte[KeyLength];
+            GetCryptoRandomBytes(key, KeyLength);
 
             // rndCharArray is expected to be 16 chars
-            char[] rndCharArray = ToBase32StringSuitableForDirName(key).ToCharArray();
+            const int RndCharArrayLength = 16;
+            char* rndCharArray = stackalloc char[RndCharArrayLength];
```

Comment:
@mellinoe Creating a string from a char pointer will create the string, pin it, and copy the data over, with a bunch of additional bounds checks and a try catch. (See `String.CtorCharPtrStartLength`) If you created with char of null & count (`CtorCharCount`) it will simply be `FastAllocateString` and a pin here.